cmake/configure/configure_symengine.cmake

```diff
@@ -0,0 +1,51 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2019 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# Configuration for the SymEngine library:
+#
+
+MACRO(FEATURE_SYMENGINE_FIND_EXTERNAL var)
+  FIND_PACKAGE(SYMENGINE)
+
+  IF(SYMENGINE_FOUND)
+    #
+    # So, we have found a SymEngine library. Let's check whether we can use it.
+    #
+    SET(${var} TRUE)
+  ENDIF()
+ENDMACRO()
+
+
+MACRO(FEATURE_SYMENGINE_CONFIGURE_EXTERNAL)
+  SET(DEAL_II_SYMENGINE_WITH_LLVM ${SYMENGINE_WITH_LLVM})
+
+  IF(DEAL_II_SYMENGINE_WITH_LLVM)
+    MESSAGE(STATUS "Configured with SymEngine LLVM capabilities.")
+  ENDIF()
+ENDMACRO()
+
+
+CONFIGURE_FEATURE(SYMENGINE)
+
+
+# 
+# Remove a flag from the list compiled by SymEngine.
+# This prevents us from compiling in C++11 mode even if we've previously 
+# determined that we will be compiling with C++14 support enabled.
+#
+REPLACE_FLAG(SYMENGINE_CXX_FLAGS_DEBUG "-std=c++11" "")
+REPLACE_FLAG(SYMENGINE_CXX_FLAGS_RELEASE "-std=c++11" "")
```

Comment:
What about you refactor this into the `MACRO_SYMENGINE_CONFIGURE_EXTERNAL` macro?